datasetinsights/io/downloader/unity_simulation.py

```diff
@@ -0,0 +1,116 @@
+import logging
+import os
+import re
+from pathlib import Path
+
+from datasetinsights.io.downloader.base import DatasetDownloader
+from datasetinsights.io.usim import Downloader, download_manifest
+
+logger = logging.getLogger(__name__)
+
+
+class UnitySimulationDownloader(DatasetDownloader):
+    """ Downloads unity simulation datasets
+        Args:
+        access_token: "Access token to be used to authenticate to
+         unity simulation for downloading the dataset"
+
+    """
+
+    PROTOCOL = "usim://"
+
+    def __init__(self, access_token=None):
+        self.access_token = access_token
+        self.run_execution_id = None
+        self.project_id = None
+
+    def download(self, source_uri, output, include_binary):
+        """
+
+        Args:
+            source_uri: This is the downloader-uri that indicates where on
+             unity simulation the dataset should be downloaded from.
+
+            output: This is the path to the directory
+            where the download will store the dataset.
+
+            include_binary: Whether to download binary files
+             such as images or LIDAR point
+            clouds. This flag applies to Datasets where metadata
+            (e.g. annotation json, dataset catalog, ...) can be separated from
+            binary files.
+
+        """
+        self.parse_source_uri(source_uri)
+        # use_cache = not args.no_cache
+        manifest_file = os.path.join(output, f"{self.run_execution_id}.csv")
+        if self.access_token:
+            manifest_file = download_manifest(
+                self.run_execution_id,
+                manifest_file,
+                self.access_token,
+                project_id=self.project_id,
+            )
+        else:
+            logger.info(
+                f"No auth token is provided. Assuming you already have "
+                f"a manifest file located in {manifest_file}"
+            )
+
+        subfolder = Path(manifest_file).stem
```

Comment:
This has hidden assumption that `<output_dir>/<run-execution-id>` is the root directly of the dataset (for `--train-data` flag in the downstream task). We should remove this assumption and make `<output_dir>` the root directory of this dataset.

Reply:
Will change Downloader to take the output directly in the constructor. For the future it might make more sense to have the output be taken in the various download methods instead of the constuctor.